test/sockets.js

```diff
@@ -1,130 +1,129 @@
+    socket.close(() => {
       t.pass('closed twice')
-      t.end()
     })
   })
 
-  socket.bind(0, function () {
+  socket.bind(0, () => {
     socket.close()
   })
 })
 
-tape('echo socket', function (t) {
-  const socket = utp()
+test('echo socket', (t) => {
+  t.plan(3)
+
+  const socket = new utp.Socket()
 
   socket.on('message', function (buf, rinfo) {
     socket.send(buf, 0, buf.length, rinfo.port, rinfo.address)
   })
 
-  socket.bind(function () {
-    var other = dgram.createSocket('udp4')
+  socket.bind(() => {
+    const other = dgram.createSocket('udp4')
     other.on('message', function (buf, rinfo) {
-      t.same(rinfo.port, socket.address().port)
-      t.same(rinfo.address, '127.0.0.1')
-      t.same(buf, Buffer.from('hello'))
+      t.is(rinfo.port, socket.address().port)
+      t.is(rinfo.address, '127.0.0.1')
+      t.alike(buf, Buffer.from('hello'))
       socket.close()
       other.close()
-      t.end()
     })
     other.send(Buffer.from('hello'), 0, 5, socket.address().port, '127.0.0.1')
   })
 })
 
-tape('echo socket with resolve', function (t) {
-  const socket = utp()
+test('echo socket with resolve', (t) => {
+  t.plan(3)
+
+  const socket = new utp.Socket()
 
   socket.on('message', function (buf, rinfo) {
     socket.send(buf, 0, buf.length, rinfo.port, 'localhost')
   })
 
-  socket.bind(function () {
+  socket.bind(() => {
     const other = dgram.createSocket('udp4')
     other.on('message', function (buf, rinfo) {
-      t.same(rinfo.port, socket.address().port)
-      t.same(rinfo.address, '127.0.0.1')
-      t.same(buf, Buffer.from('hello'))
+      t.is(rinfo.port, socket.address().port)
+      t.is(rinfo.address, '127.0.0.1')
+      t.alike(buf, Buffer.from('hello'))
       socket.close()
       other.close()
-      t.end()
     })
     other.send(Buffer.from('hello'), 0, 5, socket.address().port, '127.0.0.1')
   })
 })
 
-tape('combine server and connection', function (t) {
-  const socket = utp()
-  var gotClient = false
+test('combine server and connection', (t) => {
+  t.plan(3)
+
+  const socket = new utp.Socket()
 
   socket.on('connection', function (client) {
-    gotClient = true
-    t.same(client.remotePort, socket.address().port)
-    t.same(client.remoteAddress, '127.0.0.1')
+    t.is(client.remotePort, socket.address().port)
+    t.is(client.remoteAddress, '127.0.0.1')
     client.pipe(client)
   })
 
-  socket.listen(function () {
-    var client = socket.connect(socket.address().port)
+  socket.listen(() => {
+    const client = socket.connect(socket.address().port)
     client.write('hi')
     client.on('data', function (data) {
+      client.end()
       socket.close()
-      client.destroy()
-      t.same(data, Buffer.from('hi'))
-      t.ok(gotClient)
-      t.end()
+      t.alike(data, Buffer.from('hi'))
     })
   })
 })
 
-tape('both ends write first', function (t) {
-  var missing = 2
-  const socket = utp()
+test.skip('both ends write first', async (t) => {
```

Comment:
can we comment on this skip as well

Reply:
That one was unintentional, thanks for the heads up!